src/boundaries/stagger_and_step.jl

```diff
@@ -173,39 +173,37 @@ end
 
 
 function escape_time!(ds, x0, isinside; max_escape_time = 10000) 
-    x = copy(x0) 
-    set_state!(ds,x)
-    reinit!(ds, x)
+    set_state!(ds,x0)
+    reinit!(ds, x0)
     k = 1; 
-    while isinside(x) 
+    while isinside(current_state(ds)) 
         if k > max_escape_time 
             error("The trajectory did not escape for ", k, " steps, you probably have \
-an attractor in the defined region. Last point evaluated: ", x)
+an attractor in the defined region. Last point evaluated: ", current_state(ds))
         end
         step!(ds)
-        x = current_state(ds)
         k += 1
     end
     return current_time(ds)
 end
 
-function rand_u(δ, n, stagger_mode, rng)
+function rand_u!(u, δ, n, stagger_mode, rng)
     if stagger_mode == :exp 
         a = -log10(δ)
         s = (15-a)*rand(rng) + a
-        u = randn(rng,n)
-        u = u/norm(u)
-        return u*10.0^-s
+        u .= randn(rng,n)
+        u .= 10.0^-s*u/norm(u)
+        return
     elseif stagger_mode == :unif
         s = δ*rand(rng)
-        u = randn(rng,n)
-        u = u/norm(u)
-        return u*s
+        u .= randn(rng,n)
+        u .= s*u/norm(u)
+        return
     elseif stagger_mode == :adaptive
         s = δ*randn(rng)
-        u = randn(rng,n)
-        u = u/norm(u)
-        return u*s
+        u .= randn(rng,n)
+        u .= s*u/norm(u)
```

Comment:
`s*u` still creates a new vector. Why are you not doing simply `u .*= s/norm(u)`?

Reply:
I still have a lot to learn in optimizing Julia code. I assumed that the compiler took care of these kind of memory allocations.